src/Cech_complex/include/gudhi/Cech_complex_blocker.h

```diff
@@ -30,33 +31,109 @@ namespace cech_complex {
    *  \return true if the simplex radius is greater than the Cech_complex max_radius*/
   bool operator()(Simplex_handle sh) {
-    Point_cloud points;
-    for (auto vertex : sc_ptr_->simplex_vertex_range(sh)) {
-      points.push_back(cc_ptr_->get_point(vertex));
-#ifdef DEBUG_TRACES
-      std::clog << "#(" << vertex << ")#";
-#endif  // DEBUG_TRACES
+    using Point_cloud =  std::vector<Point_d>;
+    CGAL::NT_converter<FT, double> cast_to_double;
+    Filtration_value radius = 0.;
+
+    // for each face of simplex sh, test outsider point is indeed inside enclosing ball, if yes, take it and exit loop, otherwise, new sphere is circumsphere of all vertices
+    Sphere min_enclos_ball;
+    CGAL::NT_converter<double, FT> cast_to_FT;
+    min_enclos_ball.second = cast_to_FT(std::numeric_limits<double>::max());
+    Point_cloud face_points;
+    for (auto face : sc_ptr_->boundary_simplex_range(sh)) {
+        // Find which vertex of sh is missing in face. We rely on the fact that simplex_vertex_range is sorted.
+        auto longlist = sc_ptr_->simplex_vertex_range(sh);
+        auto shortlist = sc_ptr_->simplex_vertex_range(face);
+
+        auto longiter = std::begin(longlist);
+        auto shortiter = std::begin(shortlist);
+        auto enditer = std::end(shortlist);
+        while(shortiter != enditer && *longiter == *shortiter) { ++longiter; ++shortiter; }
+        auto extra = *longiter; // Vertex_handle
+
+        for (auto vertex : sc_ptr_->simplex_vertex_range(face)) {
+            face_points.push_back(cc_ptr_->get_point(vertex));
+    #ifdef DEBUG_TRACES
+        std::clog << "#(" << vertex << ")#";
+    #endif  // DEBUG_TRACES
+        }
+        Sphere sph;
+        auto k = sc_ptr_->key(face);
+        if(k != sc_ptr_->null_key()) {
+            sph = cc_ptr_->get_cache().at(k);
+        }
+        else {
+            sph = get_sphere(face_points.cbegin(), face_points.cend());
+        }
+        face_points.clear();
+
+        if (kernel_.squared_distance_d_object()(sph.first, cc_ptr_->get_point(extra)) <= sph.second) {
+            radius = std::sqrt(cast_to_double(sph.second));
+            #ifdef DEBUG_TRACES
+                std::clog << "center: " << sph.first << ", radius: " <<  radius << std::endl;
+            #endif  // DEBUG_TRACES
+            if (cast_to_double(sph.second) < cast_to_double(min_enclos_ball.second))
+                min_enclos_ball = sph;
+            break;
+        }
+    }
+    // Get the minimal radius of all faces enclosing balls if exists
+    if(cast_to_double(min_enclos_ball.second) != std::numeric_limits<double>::max()) {
+        radius = std::sqrt(cast_to_double(min_enclos_ball.second));
+
+        sc_ptr_->assign_key(sh, cc_ptr_->get_cache().size());
+        cc_ptr_->get_cache().push_back(min_enclos_ball);
     }
-    Filtration_value radius = Gudhi::Minimal_enclosing_ball_radius()(points);
+
+    if (radius == 0.) { // Spheres of each face don't contain the whole simplex
```

Comment:
Is it meant to be equivalent to `else` for the previous condition? Maybe the change for min_enclos_ball discussed above can help make this clearer.
And then we would likely only need a single sqrt + cast_to_double at the end instead of computing that several times.